### ISSUE TITLE
Is there a way to disconnect?

### ISSUE THREAD

**Issue by TGOlson** [mccormicka/Mockgoose#63]
I'm having a problem where my tests are hanging because of the open 'connection' to mockgoose. The tests run, pass, but never exit. If I was utilizing a live mongoose database I'd do something like this:

```
afterEach(function() {
  mongoose.connection.close();
});
```

But it doesn't appear that action does anything to help close down the process. Is there a helper method or a strategy I'm missing - or is this something that needs to be implemented?

**Comment by xavimb** [comment]
I had the same problem and I solved it removing out of the tests any modification to the lifecycle of Mongo. For some reason the `schema.pre` call was making the tests hang, but once removed it was working again.

**Comment by winfinit** [comment]
disconnect is part of 5.0 release.

**Comment by jbach** [comment]
@winfinit Has this landed? Can’t seem to find any indication in the code. So far, only `process.exit()` and Exceptions [kills the child_process](https://github.com/winfinit/mongodb-prebuilt/blob/master/index.js#L77). This leaves the process hanging, even if Mongoose.disconnect() is called.

**Comment by winfinit** [comment]
@jbach are you saying that mongod or mocha process is hanging around? and with 5.0 release?

**Comment by jbach** [comment]
I’m using [tape](https://github.com/substack/tape), so its a regular `$ node test.js`. Try this:

```JavaScript
'use strict';
let mongoose = require('mongoose');
require('mockgoose')(mongoose);

mongoose.connect('mongodb://127.0.0.1:27017/test', err => {
	if(err) return console.log(err);
	console.log('connected. should disconnect in 1s.');
});

setTimeout(() => {
	mongoose.disconnect(err =>{
		if(err) return console.log(err);
		console.log('disconnected');
	});
}, 1000);
````
I’ve played around with wrapping `mongoose.disconnect` and closing the spawned child_process in mongodb-prebuilt but the problem seems to be the [stdout listener](https://github.com/winfinit/mongodb-prebuilt/blob/master/index.js#L59). Removing the listener allows the child_process to killed properly.

**Comment by jbach** [comment]
Were you able to reproduce? FYI, I’m using Mongoose@4.2.8, Mockgoose@5.0.4 and Node@4.2.2.

**Comment by winfinit** [comment]
@jbach yes, I was able to reproduce it, thinking about a fix now

**Comment by winfinit** [comment]
@jbach just pushed disconnect hook in 5.0.9, along with some other changes that were merged in.

**Comment by jbach** [comment]
Cheers! Just fixed a small bug in https://github.com/mccormicka/Mockgoose/pull/164.

**Comment by winfinit** [comment]
thank you, just merged your fix, mockgoose@5.0.10

**Comment by dracomithril** [comment]
I still have this problem testing with JEST 
Node 8.9.1 
mongoose 4.13.7
mockgoose 7.3.3
After test are done mongod and node is still running.

Is there any solution for that?
